src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -303,6 +304,42 @@ export const handleDataSourceFetchError = (
   });
 };
 
+export interface DataSourceViewErrorWithDefaultParams {
+  changeState: (state: { showError: boolean }) => void;
+  notifications: ToastsStart;
+  failedDataSourceId: string;
+  defaultDataSource: string | null;
+  callback: () => Promise<void>;
+}
+/**
+ *
+ * @param dataSourceViewErrorParams
+ * pass in state to control the display of toast warning notification
+ * pass in the notification props to display the current error ( switch to default data source if available)
+ * pass in failedDataSourceId to construct the error message
+ * pass in the defaultDataSource to control whether display the button to switch
+ * pass in a callback if click switchToDefault button
+ */
+export const handleDataSourceViewErrorWithSwitchToDefaultOption = (
+  dataSourceViewErrorWithDefaultParams: DataSourceViewErrorWithDefaultParams
+) => {
+  const {
+    changeState,
+    notifications,
+    failedDataSourceId,
+    defaultDataSource,
+    callback,
+  } = dataSourceViewErrorWithDefaultParams;
+  changeState({ showError: true });
```

Comment:
the input {showError: true } does nothing, can we remove? @zhyuanqi found this before

Reply:
I tried to remove showError, the Button to open popover has a wired behavior, the button is first no error state, after one click, the database icon is with the red dot, then click, the popover display
![remove-showError-default](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/112784385/c531c1a0-f995-41e9-98a4-2f38c040ade1)